db/changes.go

```diff
@@ -147,19 +179,23 @@ func (db *DatabaseCollectionWithUser) AddDocToChangeEntryUsingRevCache(ctx conte
 }
 
 // Adds a document body and/or its conflicts to a ChangeEntry
-func (db *DatabaseCollectionWithUser) AddDocInstanceToChangeEntry(ctx context.Context, entry *ChangeEntry, doc *Document, options ChangesOptions) {
+func (db *DatabaseCollectionWithUser) AddDocInstanceToChangeEntry(ctx context.Context, entry *ChangeEntry, doc *Document, options ChangesOptions) error {
 
 	includeConflicts := options.Conflicts && entry.branched
 
-	revID := entry.Changes[0]["rev"]
+	revID := entry.ChangeVersionString(options.VersionType)
```

Comment:
I can't make a comment on this line, but in this function if you have a CV and `options.IncludeDocs` is specified then `get1xRevFromDoc` is called with a CV. I am not sure that all of the the code downstream from `get1xRevFromDoc` can take a CV, I am thinking `db.authorizeDoc` specifically.

Reply:
You're right - `get1xRevFromDoc` isn't going to work with CV.

I'll get test coverage of this, but maybe just queue wiring this code up behind CBG-4776 - that's required to fetch body by CV in REST API and would use same code.